crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -382,3 +382,91 @@ public macro fun zip_map_ref<$T1, $T2, $U>(
     zip_do_ref!($v1, $v2, |el1, el2| r.push_back($f(el1, el2)));
     r
 }
+
+/// Performs an in-place insertion sort on the vector `v` using the comparison function `le`.
+/// The sort is stable, meaning that equal elements will maintain their relative order.
+///
+/// Insertion sort is efficient for small vector, and can be faster than merge sort for almost
+/// sorted vectors (e.g. when the vector is already sorted or nearly sorted).
+public macro fun insertion_sort_by<$T>($v: &mut vector<$T>, $le: |&$T, &$T| -> bool) {
+    let v = $v;
+    let n = v.length();
+    if (n < 2) return ();
+    // do insertion sort
+    let mut i = 1;
+    while (i < n) {
+        let mut j = i;
+        while (j > 0 && $le(&v[j], &v[j - 1])) {
```

Comment:
Should we replace these `while`s with `do`?

Reply:
Granted the label breaks for the `$le` might be a bit annoying

Reply:
And maybe not worth it for the index math needed

Reply:
That was my motivation!

Reply:
We won't get significant bonus to readability from using macros here.